application/ui/src/features/inspect/train-model/train-model-device-picker.component.tsx

```diff
@@ -0,0 +1,116 @@
+import { useEffect, useMemo } from 'react';
+
+import { $api } from '@geti-inspect/api';
+import { Button, Content, Flex, Heading, InlineAlert, Loading, Radio, RadioGroup, Text } from '@geti/ui';
+
+import { getDeviceMetadata } from './utils/device-metadata';
+
+import classes from './train-model.module.scss';
+
+interface TrainModelDevicePickerProps {
+    selectedDevice: string | null;
+    onSelect: (device: string | null) => void;
+}
+
+export const TrainModelDevicePicker = ({ selectedDevice, onSelect }: TrainModelDevicePickerProps) => {
+    const {
+        data: availableDevices,
+        isLoading,
+        isError,
+        isRefetching,
+        refetch,
+    } = $api.useQuery('get', '/api/training-devices');
+    const devices = useMemo(() => availableDevices?.devices ?? [], [availableDevices]);
```

Comment:
I was under the impression that we wouldn’t need useMemo and useCallback anymore.